src/app/selectors.js

```diff
@@ -367,9 +370,35 @@ const countModelStatusGroups = (groupedModelStatuses) => {
   @param {Object} modelData The model data from the redux store.
   @returns {Object} The filtered model data.
 */
-const filterModelData = (filters, modelData) => {
+const filterModelData = (filters, modelData, controllerData) => {
+  const clonedModelData = cloneDeep(modelData);
+  // Add controllerName to the model data by mapping the controllerData
+  Object.entries(clonedModelData).forEach((model) => {
+    if (model[1].info) {
+      let controllerName = null;
```

Comment:
If you set the default value to the UUID then you can remove the `else` part of the conditional on 392.
```suggestion
      let controllerName = model[1].info.controllerUuid;
```

Reply:
Done